### PR TITLE
SIGTERM Handling

### PR DESCRIPTION
**Pull request by kozlov721**
Added handling of the SIGTERM signal. The current state of the training is saved and can be later resumed using the `--resume` flag.

### CODE REVIEW

File: luxonis_train/core/trainer.py [hunk 1]
```diff
@@ -56,7 +90,7 @@ def _upload_logs(self) -> None:
 
     def _trainer_fit(self, *args, **kwargs):
         try:
-            self.pl_trainer.fit(*args, **kwargs)
+            self.pl_trainer.fit(*args, ckpt_path=self.resume, **kwargs)
```
Comment (conorsim) [inline luxonis_train/core/trainer.py:93]: Unimportant for this PR, but when we resume from checkpoint, should we consider also resuming the state of some hyperparameters? I'm not sure if that is already abstracted and/or a feature of pytorch lightning.

**Review by conorsim — APPROVED** [review]

File: luxonis_train/core/trainer.py (hunk 1 above)
Comment (tersekmatija) [reply luxonis_train/core/trainer.py:93]: Yes, we should probably store all states (optimizer, scheduler, ...) so we can truly continue with the training.

Comment (kozlov721) [reply luxonis_train/core/trainer.py:93]: Yeah, pytorch lightning does this automatically.
